### PR TITLE
clean up and fix how nydus-snapshotter prepares mountpoints for nydusd

### PR DESCRIPTION
**Pull request by changweige**
1. Erofs/fscache can't work in ordinary SharedMount mode since each erofs has a seperated mountpoint while shareMode means one nydusd one mountpoint.
2. rafs as lowerdir and read-only underlying FS should be mounted at `/mnt`. `fs` is used to store bootstrap for meta layer and writeable for Active snapshot
3. `fs` has permission 0700, which causes other user in the container can't read /bin dir

### CODE REVIEW

File: snapshot/snapshot.go [hunk 1]
```diff
@@ -479,11 +479,16 @@ func (o *snapshotter) Close() error {
 }
 
 func (o *snapshotter) upperPath(id string) string {
+
```
Comment (imeoer) [inline snapshot/snapshot.go:482]: Should remove this blank line.

File: pkg/filesystem/fs/fs.go [hunk 2]
```diff
@@ -62,9 +62,15 @@ var nativeEndian binary.ByteOrder
 type Mode int
 
 const (
+	// A single nydusd serves all container images
 	SharedInstance Mode = iota
+	// One container image one nydusd topology
 	MultiInstance
+	// Rafs is integrated with other custom components.
+	// Nydusd is not needed to work as FUSE server or other backend service.
```
Comment (imeoer) [inline pkg/filesystem/fs/fs.go:70]: Nydusd is not needed to work as FUSE server (fusedev) or other fs drivers, just provide the mount slices to containerd, and do real mount by other components.

File: pkg/filesystem/fs/fs.go [hunk 3]
```diff
@@ -62,9 +62,15 @@ var nativeEndian binary.ByteOrder
...
+	// Snapshotter just cooperates with containerd with mount slice
 	NoneInstance
+	// Nydusd does not fulfill any concrete image service but prefetch data layers into local cache
```
Comment (imeoer) [inline pkg/filesystem/fs/fs.go:73]: Nydusd does not fulfill any lazy pull (on-daemon) ability but only prefetch image data into local blob cache for other components.

File: pkg/daemon/daemon.go [hunk 4]
```diff
@@ -67,14 +67,14 @@ func (d *Daemon) SharedAbsMountPoint() string {
 	return filepath.Join(*d.RootMountPoint, d.SharedMountPoint())
 }
 
+// Mountpoint of Erofs/RafsV6 and per-image nydusd/rafs
```
Comment (imeoer) [inline pkg/daemon/daemon.go:70]: Does RAFS v6 support shared mode?

File: pkg/daemon/config.go [hunk 5]
```diff
@@ -77,6 +77,8 @@ func WithLogLevel(logLevel string) NewDaemonOpt {
 	}
 }
 
+// In Shared mode, it is the mountpoint where single nydusd mounts fusedev. It has nothing
+// with fscache driver.
 func WithRootMountPoint(rootMountPoint string) NewDaemonOpt {
```
Comment (imeoer) [inline pkg/daemon/config.go:82]: We cannot assume that the fscache driver does not support multiple daemon mode, it is better not to bind the mountpoint implementation with the meaning of daemon mode.

File: snapshot/snapshot.go (hunk 1 above)
Comment (changweige) [reply snapshot/snapshot.go:482]: fixed

File: pkg/filesystem/fs/fs.go (hunk 2 above)
Comment (changweige) [reply pkg/filesystem/fs/fs.go:70]: fixed

File: pkg/filesystem/fs/fs.go (hunk 3 above)
Comment (changweige) [reply pkg/filesystem/fs/fs.go:73]: fixed

File: pkg/daemon/daemon.go (hunk 4 above)
Comment (changweige) [reply pkg/daemon/daemon.go:70]: fixed. New comment is 
// Mountpoint of per-image nydusd/rafs. It is a kernel mountpoint for each
// nydus meta layer. Each meta layer is associated with a nydusd.

File: pkg/daemon/config.go (hunk 5 above)
Comment (changweige) [reply pkg/daemon/config.go:81]: fixed

**Review by imeoer — APPROVED** [review]
LGTM, thanks for this.

**Pull request merged**